apps/api/src/app/auth/framework/roles.guard.ts

```diff
@@ -9,27 +9,39 @@ export class RolesGuard implements CanActivate {
 
   canActivate(context: ExecutionContext): boolean {
     const roles = this.reflector.get<string[]>('roles', context.getHandler());
-    if (!roles) {
+
+    if (!roles || roles.length === 0) {
+      // If no roles are specified, allow access
       return true;
     }
 
     const request = context.switchToHttp().getRequest();
-    if (!request.headers.authorization) return false;
-
-    const token = request.headers.authorization.split(' ')[1];
-    if (!token) return false;
-
     const authorizationHeader = request.headers.authorization;
-    if (!authorizationHeader?.includes('ApiKey')) {
-      const user = jwt.decode(token) as IJwtPayload;
-      if (!user) return false;
-    }
 
-    /*
+    if (!authorizationHeader || !authorizationHeader.startsWith('Bearer ')) {
+      throw new UnauthorizedException('Unauthorized');
+    }
+     /*
```

Comment:
This change breakes the authentication for our API. I am sorry but we have to reject it.

Reply:
Can we remove that line ?